app/src/main/java/ai/haptik/android/sample/app/MyFirebaseMessagingService.java

```diff
@@ -12,18 +13,18 @@
     @Override
     public void onMessageReceived(RemoteMessage remoteMessage) {
         final Map<String, String> data = remoteMessage.getData();
-        if (NotificationManager.isHaptikNotification(data)) {
+        if (HaptikNotificationManager.isHaptikNotification(data)) {
             if (!HaptikLib.isInitialized()) {
                 Handler mainHandler = new Handler(getApplicationContext().getMainLooper());
                 mainHandler.post(new Runnable() {
                     @Override
                     public void run() {
-                        HaptikLib.init(Utils.getHaptikInitData(getApplication()));
-                        NotificationManager.handleNotification(getApplicationContext(), data);
+                        HaptikLib.init(((InitDataCallback) getApplicationContext()).getClientSetupData());
```

Comment:
Keep this as it is. `InitDataCallback` is for Haptik SDK's usage and not needed by clients, since they already know where to ge the `InitData` from

Reply:
Done